jmxtrans-core/src/main/java/com/googlecode/jmxtrans/model/naming/KeyUtils.java

```diff
@@ -22,12 +22,12 @@
  */
 package com.googlecode.jmxtrans.model.naming;
 
+import com.google.common.base.Joiner;
 import com.googlecode.jmxtrans.model.Query;
 import com.googlecode.jmxtrans.model.Result;
 import com.googlecode.jmxtrans.model.Server;
 
 import java.util.List;
-import java.util.Map;
 
 public final class KeyUtils {
```

Comment:
Note for later: If we get a clean abstraction on metrics names (and you are starting that way with introducing valuePath), then we can start converting this utility class to a naming strategy. The "." separated string is what a lot of OutputWriter need, but it is far from universal. And having this in a static class makes it harder to compose...

Reply:
I thought about adding a getValuePathAsString method in Result class, because at the moment almost all writer implementations use this valuePath as dot delimited String. At the end I placed this code in KeyUtils class.

Reply:
Yep, `KeyUtils` look like the right place to me too. At least until we refactor this a bit more.